pkg/storage/etcd/testing/utils.go

```diff
@@ -124,6 +126,22 @@ func (m *EtcdTestServer) launch(t *testing.T) error {
 	return nil
 }
 
+// waitForEtcd wait until etcd is propagated correctly
+func (m *EtcdTestServer) waitUntilUp() error {
+	membersAPI := etcd.NewMembersAPI(m.Client)
+	for start := time.Now(); time.Since(start) < 5*time.Second; time.Sleep(10 * time.Millisecond) {
+		members, err := membersAPI.List(context.TODO())
+		if err != nil {
+			glog.Errorf("Error when getting etcd cluster members")
+			continue
+		}
+		if len(members) == 1 && len(members[0].ClientURLs) > 0 {
```

Comment:
I did it consciously - I think that if ClientURLs and there is one members (which is exactly what is happening at the beginning), I want to return a timeout.